### PR TITLE
Cleanup and update

### PR DESCRIPTION
**Pull request by lukasmatusiewicz**
This pull request includes several changes to the `privacyIDEA` provider for Keycloak, focusing on documentation updates, dependency upgrades, and codebase simplifications by removing deprecated functionalities and improving code consistency.

### Documentation Updates:
* Updated references from "2FA" to "MFA" in `README.md` and added punctuation for consistency.
* Deprecated and removed certain configuration parameters related to token enrollment and preferred token type in `README.md`. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L33-L37) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R45-L50)

### Dependency Upgrades:
* Upgraded `keycloak-core` dependency from version `23.0.4` to `24.0.3` in `pom.xml`.
* Upgraded `okhttp` dependency from version `4.9.1` to `5.0.0-alpha.3` in `pom.xml`.

### Codebase Simplifications:
* Removed deprecated configuration constants and related code from `Configuration.java` and `Const.java`. [[1]](diffhunk://#diff-321994e6a006fd379d84cf9e38c4aa371867c099d3dcae684b661f9c13642892L27-L35) [[2]](diffhunk://#diff-7132c091aaa79d9b66c486e636a15bd2ab23eb167f1ff15632765435de5fec92L47)
* Cleaned up imports and removed unused classes and constants in `PrivacyIDEAAuthenticator.java`. [[1]](diffhunk://#diff-0e8cdfe80e57354c310e68c2cc4c27f9b3caac83347d0e211e9f9aff6ca10c3aL48-L54) [[2]](diffhunk://#diff-0e8cdfe80e57354c310e68c2cc4c27f9b3caac83347d0e211e9f9aff6ca10c3aL74-L82) [[3]](diffhunk://#diff-0e8cdfe80e57354c310e68c2cc4c27f9b3caac83347d0e211e9f9aff6ca10c3aL91-R86)

These changes aim to enhance the maintainability and clarity of the codebase while ensuring the documentation is up-to-date with the latest features and best practices.

### CODE REVIEW

**Comment by lukasmatusiewicz** [comment]
last tests...

**Comment by lukasmatusiewicz** [comment]
closing #187

File: src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticatorFactory.java [hunk 1]
```diff
@@ -101,25 +102,13 @@ public List<ProviderConfigProperty> getConfigProperties()
...
-        configProperties.add(piPrefToken);
-
         ProviderConfigProperty piDoTriggerChallenge = new ProviderConfigProperty();
         piDoTriggerChallenge.setType(ProviderConfigProperty.BOOLEAN_TYPE);
         piDoTriggerChallenge.setName(Const.CONFIG_TRIGGER_CHALLENGE);
         piDoTriggerChallenge.setLabel("Enable trigger challenge");
         piDoTriggerChallenge.setHelpText(
                 "Choose if you want to trigger challenge-response token using the provided service account before the second step of authentication. " +
-                "This setting is mutually exclusive with sending any password and will take precedence over both.");
+                        "This setting is mutually exclusive with sending any password and will take precedence over both.");
```
Comment (nilsbehlen) [inline src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticatorFactory.java:111]: the formatting is off here

**Review by nilsbehlen — CHANGES_REQUESTED** [review]

File: src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticator.java [hunk 2]
```diff
@@ -149,8 +113,7 @@ private Pair loadConfiguration(final AuthenticationFlowContext context)
      *
      * @param context AuthenticationFlowContext
      */
-    @Override
-    public void authenticate(AuthenticationFlowContext context)
+    @Override public void authenticate(AuthenticationFlowContext context)
```
Comment (nilsbehlen) [inline src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticator.java:116]: this is wrong formatting

Comment (lukasmatusiewicz) [reply src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticator.java]: [0bd9b91](https://github.com/privacyidea/keycloak-provider/pull/195/commits/0bd9b91d93771b6f1578322981dc4b3b60a83104)

**Review by nilsbehlen — APPROVED** [review]

**Pull request merged**